### PR TITLE
5322: GET - order description

### PR DESCRIPTION
**Pull request by RuthBerry96**
[AB#6892](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6892)

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderDescriptionControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+        [Test]
+        public async Task Get_OrderIdExists_ReturnsTheOrdersDescription()
+        {
+            var orderId = "C0000014-01";
+
+            var testData = CreateOrderDescriptionTestData(orderId, "Test Description");
+
+            var context = OrderDescriptionTestContext.Setup();
+            context.Order = testData.order;
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderDescriptionControllerTests.cs:47]: I find the lowercase `order` a little strange, but have the opposite when creating the tuple. Could be one for the coding standards meeting as I am not sure what to suggest & it is very minor.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Get Description 5322.feature [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Background:
+	Given Orders exist
+		| OrderId    | Description         | OrderStatusId | Created    | LastUpdated | LastUpdatedBy                        | OrganisationId                       |
+		| C000014-01 | Some Description    | 1             | 11/05/2020 | 11/05/2020  | 335392e4-4bb1-413b-9de5-36a85c9c0422 | 4af62b99-638c-4247-875e-965239cd0c48 |
+		| C000014-02 | Another Description | 2             | 05/05/2020 | 09/05/2020  | a11a46f9-ce6f-448a-95c2-fde6e61c804a | 4af62b99-638c-4247-875e-965239cd0c48 |
+
+@5322
+Scenario: 1. Get an orders description
+	When a GET request is made for an orders description with orderId C000014-01
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Get Description 5322.feature:14]: Naming stuff is difficult & I tend to struggle with this, but I believe spec flow scenarios should try to represent business requirements. This means less "technical" language e.g orderId. So for example you this be more readable to a BA:

When the user makes a request to retrieve the order description section with the ID <>

Found this [link](https://automationpanda.com/2017/01/30/bdd-101-writing-good-gherkin/) as a good resource.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderDescriptionSteps.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+            var actual = new OrderDescriptionTable()
+            {
+                Description = response.SelectToken("description").ToString()
+            };
+
+            actual.Should().BeEquivalentTo(expected);
+        }
+
+        public sealed class OrderDescriptionTable
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderDescriptionSteps.cs:49]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderDescriptionControllerTests.cs (hunk 1 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderDescriptionControllerTests.cs:47]: Lets discuss it on Friday, as tbh I'm not sure what the best way to do it is. If you want me to change it just say 😄

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Get Description 5322.feature (hunk 2 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Get Description 5322.feature:14]: 👍

**Pull request merged**